syntax = "proto2";

message Empty{}

service Card {
  // Sends a greeting
  rpc getCards (Empty) returns (IdCardsResponse) {}
  // Sends another greeting
  rpc getFullCard (SimpleCard) returns (FullCard) {}
}

// The response message containing the greetings
message SimpleCard {
  required int32 id = 1;
}

message FullCard {
  required string id = 1;
  required string name = 2;
  required string health = 3;
  required string attack = 4;
  required string defense = 5;
}

message IdCardsResponse { // LIST
  repeated int64 result = 1;
}